src/Pos/Communication/Limits.hs

```diff
@@ -92,6 +96,19 @@ instance MessageLimitedPure (AbstractHash Blake2b_224 a) where
 instance MessageLimitedPure (AbstractHash Blake2b_256 a) where
     msgLenLimit = 32
 
+instance MessageLimitedPure w => MessageLimitedPure (ProxySecretKey w) where
+    msgLenLimit = coerce (msgLenLimit @w)
+                    + 2 * (coerce (msgLenLimit @PublicKey))
+                    + (coerce (msgLenLimit @(Signature w)))
+
+instance MessageLimitedPure w => MessageLimitedPure (ProxySignature w a) where
+    msgLenLimit = coerce (msgLenLimit @w)
+                    + (coerce (msgLenLimit @PublicKey))
+                    + 2 * (coerce (msgLenLimit @(Signature w)))
+
+instance MessageLimitedPure EpochIndex where
+    msgLenLimit = 8
```

Comment:
Are you sure it's correct? I guess it's not. See comments to `MessageLimitedPure`. Easy way to find a limit is to add a test.